### PR TITLE
Phantasmic Dream Fluff:Zekes Tendrils

### PR DESCRIPTION
**Pull request by Aurorablade**
ckey:phantasmicdream
Character: Zeke Varloss (i probaly did not do the last name right sorry)
Usage:
One Skrell Hair of longer tentacles
Pictured: https://puu.sh/wbPn8/52a1ddc540.png

Hair aquired via hand mirror,  with inhand sprites (its in the picture i just didn't get a side view)

This hairstyle will need to be locked out to normal players if code does not already do that. Please advice if i have to fix that on my end.

Heratic chruch was the one who got the code, i just made sure it worked as code is voodoo to churchy.

### CODE REVIEW

**Comment by PhantasmicDream** [comment]
Zeke Varloss is the spelling. 

Thank you for doing this!

**Comment by Fox-McCloud** [comment]
Don't put the in-hand items with the rest; they have their own unique files.

**Comment by Aurorablade** [comment]
@Fox-McCloud i sometimes lose what sprite goes where..whats the right one for this?

File: code/modules/customitems/item_defines.dm [hunk 1]
```diff
@@ -960,6 +960,26 @@
...
+	throwforce = 0
+	w_class = 2
+
+/obj/item/fluff/zekemirror/attack_self(mob/user)
+	var/mob/living/carbon/human/target = user
+	if(!istype(target) || target.get_species() != "Skrell") // It'd be strange to see other races with head tendrils.
+		return
+
+	if(target.change_hair("Zeke's Tentacles"))
+		to_chat(target, "<span class='notice'>You take time to admire yourself in the [src], brushing your tendrils down and revealing their true length.</span>")
```
Comment (Fox-McCloud) [inline code/modules/customitems/item_defines.dm]: Please stop using `\the [src]` or `the [src]`

Comment (Aurorablade) [inline code/modules/customitems/item_defines.dm]: I actually got this from dreamy and didn't notice it myself..sorrysorry..

File: code/modules/customitems/item_defines.dm [hunk 2]
```diff
@@ -960,6 +960,26 @@
...
+/obj/item/fluff/zekemirror //phantasmicdream : Zeke Varloss
+	name = "engraved hand mirror"
+	desc = "A very classy hand mirror, with fancy detailing."
+	icon = 'icons/obj/custom_items.dmi'
+	icon_state = "hand_mirror"
+	attack_verb = list("smacked")
+	hitsound = 'sound/weapons/tap.ogg'
+	force = 0
+	throwforce = 0
+	w_class = 2
```
Comment (SamHPurp) [inline code/modules/customitems/item_defines.dm]: please use the weight define `small`

**Comment by taukausanake** [comment]
Don't look tentaclely enough. Needs more imply-able hair rape.

I was really hoping this would look like the Skrell tentacle GIF that was shared once. I am very disappointed :(

**Comment by HereticChurch** [comment]
This should probably be marked DNM until the hairstyle is properly locked to other players. I couldn't figure that bit out.

Also, for the changelog my name is HereticChurch.

**Comment by KasparoVy** [comment]
@Aurorablade You could define a 'fluff' hair subtype like `/datum/sprite_accessory/hair/fluff` and give it a var like `owner_ckey`, then change your definition for Zeke's hairstyle to be a subtype and write in PhantasmicDream for the owner_ckey and set the species as "Skrell" so it'll restrict properly. Then just add the appropriate logic for fluff hair in the places where sprite accessory lists are compiled and that'd be all.

**Comment by Aurorablade** [comment]
I got to fix conflicts later but this is probaly good to go on virtue of i think it can be slected unless it has an allowe species list. i am going to double check.

**Comment by Aurorablade** [comment]
@KasparoVy give em a further example so i can double check what i am thinking about useing.

File: code/modules/mob/new_player/sprite_accessories.dm [hunk 3]
```diff
@@ -927,6 +934,15 @@
 	icon_state = "nuc_neutron"
 
 
+
+/datum/sprite_accessory/hair/fluff
```
Comment (Crazylemon64) [inline code/modules/mob/new_player/sprite_accessories.dm]: Add the attribute `fluff = 1` here so you don't need to re-declare it for every fluff style

**Comment by Aurorablade** [comment]
Works BTW just got to fix conflicts

**Comment by tigercat2000** [comment]
No changelog please.

**Comment by taukausanake** [comment]
Still think it would be nice to have fluff additions added to the changelogs. It wouldn't hurt either way; it's not like anyone reads them

**Comment by Aurorablade** [comment]
changelog removed. Code suggestiong in progress

File: code/modules/mob/new_player/sprite_accessories.dm (hunk 3 above)
Comment (Aurorablade) [inline code/modules/mob/new_player/sprite_accessories.dm]: it already is added on my end...

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 4]
```diff
@@ -21,7 +21,7 @@
 
 	gender = new_gender
 
-	var/datum/sprite_accessory/hair/current_hair = hair_styles_list[H.h_style]
+	var/datum/sprite_accessory/hair/current_hair = hair_styles_public_list[H.h_style]
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/appearance.dm]: Wrong, use hair_styles_full_list.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 5]
```diff
@@ -442,7 +442,7 @@ var/global/list/damage_icon_parts = list()
 	//var/icon/debrained_s = new /icon("icon"='icons/mob/human_face.dmi', "icon_state" = "debrained_s")
 
 	if(head_organ.h_style && !(head && (head.flags & BLOCKHEADHAIR) && !(isSynthetic())))
-		var/datum/sprite_accessory/hair/hair_style = hair_styles_list[head_organ.h_style]
+		var/datum/sprite_accessory/hair/hair_style = hair_styles_full_list[head_organ.h_style]
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Wrong, use hair_styles_full_list.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 6]
```diff
@@ -759,7 +759,7 @@ var/global/list/damage_icon_parts = list()
 		else
 			new_glasses = image("icon" = 'icons/mob/eyes.dmi', "icon_state" = "[glasses.icon_state]")
 
-		var/datum/sprite_accessory/hair/hair_style = hair_styles_list[head_organ.h_style]
+		var/datum/sprite_accessory/hair/hair_style = hair_styles_full_list[head_organ.h_style]
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Wrong, use hair_styles_full_list.

File: code/modules/mob/new_player/preferences_setup.dm [hunk 7]
```diff
@@ -341,7 +341,7 @@
 		face_s.Blend(eyes_s, ICON_OVERLAY)
 
 
-	var/datum/sprite_accessory/hair_style = hair_styles_list[h_style]
+	var/datum/sprite_accessory/hair_style = hair_styles_public_list[h_style]
```
Comment (tigercat2000) [inline code/modules/mob/new_player/preferences_setup.dm]: Wrong, use hair_styles_full_list.

**Comment by tigercat2000** [comment]
LGTM

**Pull request merged**